### PR TITLE
Fix calendar portlet to work when selected item is a Collection

### LINKED ISSUE #399: Calendar portlet breaks when selecting a Collection with date restriction as a source
We have a collection to show future events. The collection is configured to show objects of type event that end in the future (the corresponding catalog query would be `end={'query': DateTime(), 'range': 'start'}`

If we configure a calendar portlet and select the said collection as source, it breaks with the following traceback

```
[2024-07-18 11:40:40,098] ERROR:Zope.SiteErrorLog:waitress-1:TypeError: http://localhost:8080/Plone/eu/agenda/aggregator/agenda-view
Traceback (innermost last):
  Module plone.app.portlets.manager, line 69, in safe_render
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 58, in __call__
  Module zope.pagetemplate.pagetemplate, line 134, in pt_render
  Module Products.PageTemplates.engine, line 365, in __call__
  Module z3c.pt.pagetemplate, line 174, in render
  Module chameleon.zpt.template, line 331, in render
  Module chameleon.template, line 217, in render
  Module chameleon.utils, line 20, in raise_with_traceback
  Module chameleon.template, line 193, in render
  Module 97a686d311be2171d2acfc4a8ce65ee1, line 614, in render
  Module zope.tales.expressions, line 248, in __call__
  Module Products.PageTemplates.Expressions, line 213, in _eval
  Module zope.tales.expressions, line 153, in _eval
  Module Products.PageTemplates.Expressions, line 130, in trustedBoboAwareZopeTraverse
  Module zope.traversing.adapters, line 149, in traversePathElement
   - __traceback_info__: (<plone.app.event.portlets.portlet_calendar.Renderer object at 0x792102796b10>, 'cal_data')
  Module zope.traversing.adapters, line 46, in traverse
   - __traceback_info__: (<plone.app.event.portlets.portlet_calendar.Renderer object at 0x792102796b10>, 'cal_data', [])
  Module plone.app.event.portlets.portlet_calendar, line 236, in cal_data
TypeError: '<' not supported between instances of 'dict' and 'datetime.date'

 - Expression: "view/cal_data"
 - Filename:   ... -py3.11.egg/plone/app/event/portlets/portlet_calendar.pt
 - Location:   (line 85: col 31)
 - Source:     <tr tal:repeat="week view/cal_data">
                                    ^^^^^^^^^^^^^
 - Arguments:  template: <Products.Five.browser.pagetemplatefile.ViewPageTemplateFile object at 0x7920f83a6cd0>
               options: {}
               args: ()
               nothing: None
               modules: <Products.PageTemplates.ZRPythonExpr._SecureModuleImporter object at 0x79210e538290>
               request: <WSGIRequest, URL=http://localhost:8080/Plone/eu/agenda/aggregator/agenda-view>
               view: <plone.app.event.portlets.portlet_calendar.Renderer object at 0x792102796b10>
               context: <Collection at /Plone/eu/agenda/aggregator>
               views: <Products.Five.browser.pagetemplatefile.ViewMapper object at 0x7920f833b690>
               here: <Collection at /Plone/eu/agenda/aggregator>
               container: <Collection at /Plone/eu/agenda/aggregator>
               root: <Application at >
               traverse_subpath: []
               user: <PropertiedUser 'admin2'>
               default: <DEFAULT>
               repeat: <Products.PageTemplates.engine.RepeatDictWrapper object at 0x7920f830f1b0>
               loop: {'weekday': <Products.PageTemplates.engine.RepeatItem object at 0x7920f83389d0>}
               target_language: None
               translate: <function BaseTemplate.render.<locals>.translate at 0x7920f83159e0>
               attrs: {}
               navigation_root_url: 'http://localhost:8080/Plone/eu'
               month: 7
               year: 2024


```

### PR DESCRIPTION
**Pull request by erral**
Fixes #399

### CODE REVIEW

File: plone/app/event/portlets/portlet_calendar.py [hunk 1]
```diff
@@ -26,8 +26,10 @@
 from zope.component.hooks import getSite
 from zope.i18nmessageid import MessageFactory
 from zope.interface import implementer
+from DateTime import DateTime
```
Comment (stevepiercy) [inline plone/app/event/portlets/portlet_calendar.py:29]: I think this needs to be moved to line 3, according to isort.

**Review by stevepiercy — DISMISSED** [review]

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

File: plone/app/event/portlets/portlet_calendar.py [hunk 2]
```diff
@@ -211,12 +211,9 @@ def cal_data(self):
...
 
-            # restrict start/end with those from query, if given.
-            if "start" in query and query["start"] > start:
-                start = query["start"]
-            if "end" in query and query["end"] < end:
-                end = query["end"]
-
+            # When having a collection as context, we should use the configured query
+            # except the start and the end, which should be
+            # taken from the current calendar start and end
```
Comment (stevepiercy) [inline plone/app/event/portlets/portlet_calendar.py:216]: Will the removal of the restriction still apply for a query for events that are _not_ in a collection as context?

**Review by stevepiercy — COMMENTED** [review]
Just a sanity check question. Otherwise LGTM, but this should be reviewed by someone else who has more familiarity with the code than I. Would you please request from other reviewers?

File: plone/app/event/portlets/portlet_calendar.py (hunk 2 above)
Comment (davisagli) [reply plone/app/event/portlets/portlet_calendar.py:216]: It never did. This code was before and still is inside a conditional that checks for the context being a collection.

**Review by davisagli — APPROVED** [review]

**Pull request merged**